test/jasmine/tests/sankey_test.js

```diff
@@ -601,6 +609,30 @@ describe('sankey tests', function() {
             .catch(failTest)
             .then(done);
         });
+
+        it('should not output hover/unhover event data when hovermoder is false', function(done) {
+            var fig = Lib.extendDeep({}, mock);
+
+            Plotly.plot(gd, fig)
+            .then(function() { return Plotly.relayout(gd, 'hovermode', false); })
+            .then(function() { return _hover('node'); })
+            .then(failTest).catch(function(err) {
+                expect(err).toBe('plotly_hover did not get called!');
+            })
```

Comment:
Ah, clever 🎉 
Does `failTest` give an intelligible message if we get there? ie if we run this test without your previous commit, would it be clear what went wrong?

Reply:
It returns: `Chrome 68.0.3440 (Windows 10 0.0.0) sankey tests Test hover/click event data: should not output hover/unhover event data when hovermoder is false FAILED`
followed by:
`Expected Object({ event: [object MouseEvent], points: [ Object({ pointNumber: 4, label: 'Solid', color: 'rgba(148, 103, 189, 0.8)', sourceLinks: [ Object({ pointNumber: 60, label: '', color: 'rgba(0,0,96,0.2)', source: <circular reference: Object>, target: Object({ pointNumber: 26, label: 'Thermal generation', color: 'rgba(227, 119, 194, 0.8)', sourceLinks: [ Object({ pointNumber: 63, label: '', color: 'rgba(0,0,96,0.2)', source: <circular reference: Object>, target: Object, value: 787.129, originalIndex: 63, dy: 84.76807405050613, ty: 0, sy: 0, trace: Object, curveNumber: 0 }), Object({ pointNumber: 62, label: '', color: 'rgba(0,0,96,0.2)', source: <circular reference: Object>, target: Object, value: 525.531, originalIndex: 62, dy: 56.595870211663566, ty: 0, sy: 84.76807405050613, trace: Object, curveNumber: 0 }), Object({ pointNumber: 64, label: '', color: 'rgba(0,0,96,0.2)', source: <circular reference: Object>, target: Object, value: 79.329, originalIndex: 64, dy: 8.543156898491352, ty: 0,  ... to be undefined.`

What would be preferred?

Reply:
That looks great, assuming there's a line number along with it too, so you can tell it's the `_hover` on line 618 that caused the failure.

Reply:
(no it doesn't give a line number, but neither do our normal `failTest` invocations... lets leave it as is at least for now)